core/src/com/mygdx/game/client/screen/GameScreen.java

```diff
@@ -111,4 +87,21 @@ public void resize(int width, int height) {
   public void dispose() {
     renderer.dispose();
   }
+
+  private void setUpGameState() {
+    gameState.setFields(mapInitializer.initializeMap());
+    startUnitInitializer.initializeTestUnit(gameState.getFields());
+  }
+
+  private void setUpInput() {
+    var cameraInputProcessor = new CameraMoverInputProcessor(viewport);
```

Comment:
może to wynieść wyżej i zrobić final

Reply:
Z trzech opcji (lokalna konstrukcja, finalna na całą klasę, wstrzykiwanie) to najlepsze wydaje mi się wstrzykiwanie, a właśnie najmniej sensowne wynoszenie tego do większego scope'a. Jakbym miał to teraz zmieniać to zrobiłbym chyba nawet całego InputMultiplexera wstrzykiwanego, ale w sumie tyle jest tego wstrzykiwania, że zaczynam wątpić czy to dobre podejście xD

Jakby, udało mi się dalej zredukować liczbę parametrów do gamescreen do kilku tylko, ale przez to GameState jest wstrzykiwany tylko do jakiegoś InputAdaptera i to trochę dziwnie wygląda tbh.